### ISSUE TITLE
Should bin/generate be a rake task?

### ISSUE THREAD

**Issue by tommyschaefer** [exercism/xruby#536]
Do you all think `bin/generate` should be converted to a Rake task?

Without having thought about it too much, I would imagine the interface could be similar to the recent `rake test` stuff. For example:

```
rake generate:hamming -- -f
```

Pros:
- Consistency with other repository "tools" (`rake test`)
- Can be called from subdirectories of `xruby` without `../../bin/generate ...`
- Can easily be chained `rake generate:hamming test:hamming`

Cons:
- Options are more complicated (`-- -f` vs `-f`)
- Options are shared by all running rake tasks. You may want to run `generate:hamming` with `-v`, but not the tests. Not too hard to add per-task args like `Rake::TestTask` (`TESTOPTS=''` / `GENOPTS=''`), but adds complexity.

I'm not sure if I have a strong opinion either way, but I figured I'd ask the question anyway!

Thanks for your time!

**Comment by kotp** [comment]
I don't change to sub directories of the project... so was not aware that there was (still) a problem with calling the bin/generate while under a sub directory.

The options are more complicated, but fall in line with current and historical unix arguments... so I don't consider this a "con".

Having this as a rake task I think is a good thing.  It doesn't mean that you can't still run it with the bin/generate directly.

It already can be chained with the shell being used, but this is a reasonable thing as well.

In summary: Not sure that it should be done, but it could be done.  I don't think there is a driving need for it at the moment?  Probably wait for some time to pass on the current rake implementation to be tried and true.

**Comment by tommyschaefer** [comment]
> In summary: Not sure that it should be done, but it could be done. I don't think there is a driving need for it at the moment? Probably wait for some time to pass on the current rake implementation to be tried and true.

Yeah, that's what I was thinking as well. :+1:

Should I go ahead and close the issue since it's not something we're thinking about now (and we can reference the closed issue later), or is it better to keep it open as a reminder to think about in the future?

**Comment by kotp** [comment]
Closing, but could be opened again if wanted.
